script/arrays.js

```diff
@@ -0,0 +1,221 @@
+        }
+    }
+}
+
+shopListOrdered(shopingList);
+
+//  Покупка продукту. Функція приймає назву продукту і відзначає його як придбаний.
+
+productPurchase = (productName) => {
+    for (let i = 0; i < shopingList.length; i++) {
+        if (shopingList[i].name == productName && shopingList[i].bought == true) {
+            console.log(`You have already bought ${productName}.`);
+            break;
+        }
+        if (shopingList[i].name == productName && shopingList[i].bought == false) {
+            shopingList[i].bought = true;
+            console.log(`You have successfully purchased ${productName}.`);
+            break;
+        } else {
+            console.log(`${productName} is not in the shopping list.`);
+        }
+    }
+}
+
+//  Створення списку (не) придбаних продуктів.
+
+shopListUnpurchased = (arr) => {
+    console.log("Shopping List of Unpurchased Products");
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == false) {
+            console.log(arr[i]); 
+            // console.log(arr[i].name); - якщо треба тільки назви продуктів вивести
+        } 
+    }
+}
+
+productPurchase("apple");
+shopListUnpurchased(shopingList);
+
+
+// standart
+
+
+// 1. Видалення продукту зі списку (видалення повинно проводитися шляхом створення нового масиву, 
+// в якому продукт, що ми шукаємо, буде відсутнім)
+
+productRemove = (productName) => {
+    shopingList = shopingList.map((item) => {
+        if (item.name != productName) {
+            return item;
+        }
+    });
+
+    for (let i = 0; i < shopingList.length; i++) {
+        if (shopingList[i] == undefined) {
+            shopingList.splice(i, 1);
+        } 
+    }
+    return shopingList;
+}
+
+console.log(shopingList);
+productRemove("apple");
+console.log(shopingList);
+productRemove("cheese");
+console.log(shopingList);
+
+
+// 2. Додавання покупки в список. Враховуй, що при додаванні покупки з уже існуючим в списку продуктом, 
+// необхідно збільшувати кількість в існуючій покупці, а не додавати нову. При цьому також повинна змінитися сума, 
+// наприклад, якщо ціна за одиницю 12, а кількості товарів стало 2, то сума буде 24.
+
+function isPositiveNumber (a) {
+    if (a > 0 && a == +a) {
+        return true;
+    } else {
+        alert('This number is not a positive number.')
+        return false;
+    }
+}
+
+productAdd = (productName) => {
+    let productQuantity;
+    let productPriceUnit;
+    let check;
+    do {
+        productQuantity = prompt(`Enter the number of ${productName}s you want to buy.`);
+        check = isPositiveNumber(productQuantity);
+    } while (check == false)
+
+    for (let i = 0; i < shopingList.length; i++) {
+        if (shopingList[i].name == productName && shopingList[i].bought == false) {
+            shopingList[i].quantity = shopingList[i].quantity + +productQuantity;
+            return console.log(`The ${productName} is already on the shopping list. Their number has increased by ${productQuantity}.`);
```

Comment:
`return console.log(...)` will exit from the function with the result `undefined`. If you want to drop out from function just use return in next new line `return;`

In real-world we do not to throw or save the result of the `console.log` function